### PR TITLE
Feat/upload blob vercel

### PR DESCRIPTION
**Pull request by jackhuynh95**
https://github.com/naveennaidu/novel-vue/assets/26354519/ab1ed7d5-1984-4fe8-a5fd-149c7959d568

### CODE REVIEW

File: package/src/components/plugins/upload-images.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+export const handleImageUpload = (file: File) => {
+  // upload to Vercel Blob
+  return new Promise((resolve) => {
+    toast.promise(
+      fetch("/api/upload", {
+        method: "POST",
+        headers: {
+          "content-type": file?.type || "application/octet-stream",
+          "x-vercel-filename": file?.name || "image.png",
```
Comment (naveennaidu) [inline package/src/components/plugins/upload-images.ts]: Not able to upload with image name `Screenshot 2023-10-23 at 10.13.48 AM`

https://github.com/naveennaidu/novel-vue/assets/28999143/75764db8-822f-47b2-b435-8b2ad80f236c

You can fix this by using `encodeURIComponent(file?.name || "image.png")`

File: package/src/components/plugins/upload-images.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+      .setMeta(uploadKey, { remove: { id } });
+    view.dispatch(transaction);
+  });
+}
+
+export const handleImageUpload = (file: File) => {
+  // upload to Vercel Blob
+  return new Promise((resolve) => {
+    toast.promise(
+      fetch("/api/upload", {
```
Comment (naveennaidu) [inline package/src/components/plugins/upload-images.ts]: What do you think about providing this image upload endpoint as prop?

**Review by naveennaidu — CHANGES_REQUESTED** [review]
Hey, Thank you for adding image support. Really appreciate it. I have minor improvement and a question.

File: package/src/components/plugins/upload-images.ts (hunk 2 above)
Comment (jackhuynh95) [reply package/src/components/plugins/upload-images.ts]: Yeah, That is okay. let do that

File: package/src/components/plugins/upload-images.ts (hunk 1 above)
Comment (jackhuynh95) [reply package/src/components/plugins/upload-images.ts]: you can show me **devtools -> network status**

Comment (naveennaidu) [reply package/src/components/plugins/upload-images.ts]: Not making any network call. this error is because of name encoding issue

Comment (jackhuynh95) [reply package/src/components/plugins/upload-images.ts]: I got it, and this issue is resolved

File: package/src/components/plugins/upload-images.ts (hunk 2 above)
Comment (jackhuynh95) [reply package/src/components/plugins/upload-images.ts]: https://github.com/naveennaidu/novel-vue/pull/4/commits/9e364c210b9db3f6327a10fd39b41598c35ecaab and https://github.com/naveennaidu/novel-vue/pull/4/commits/ac611262b76ee4cf57d586c3d985a55ed9790dc0

=> Enhance in here, Can you review these commits

Comment (naveennaidu) [reply package/src/components/plugins/upload-images.ts]: Looks good!

**Comment by naveennaidu** [comment]
There is a conflict. Can you pull latest main branch and resolve the conflict? I can merge the PR to main

**Review by naveennaidu — APPROVED** [review]

**Pull request closed without merging**